pkg/cluster/addressmap.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import "sync"
+
+// AddressMap provides a thread-safe mapping of addresses for each Kubernetes cluster.
+type AddressMap struct {
+	// Addresses hold the underlying map. Visible only for testing, for the purposes of simplified construction.
+	// Production code should never access this directly.
+	Addresses map[ID][]string
+
+	// NOTE: The copystructure library is not able to copy unexported fields, so the mutex will not be copied.
+	mutex sync.RWMutex
```

Comment:
You can add a tag ``copy:"ignore"``, so that the mutex will be ignored when copied

Reply:
It's not needed. Private fields are not copied by copystructure.